pkg/pillar/cmd/volumemgr/handlework.go

```diff
@@ -156,10 +156,27 @@ func volumeWorker(ctxPtr interface{}, w worker.Work) worker.WorkResult {
 	var volumeCreated bool
 	var fileLocation string
 	var err error
+	vcp := types.VolumeCreatePending{
+		VolumeID:               d.status.VolumeID,
+		GenerationCounter:      d.status.GenerationCounter,
+		LocalGenerationCounter: d.status.LocalGenerationCounter,
+		ContentFormat:          d.status.ContentFormat,
+		Encrypted:              d.status.Encrypted,
+	}
 	if d.create {
+		//set pending create operation
+		_ = ctx.pubVolumeCreatePending.Publish(vcp.Key(), vcp)
 		volumeCreated, fileLocation, err = createVolume(ctx, d.status)
+		if err == nil {
+			//in case of no error remove pending create operation
+			_ = ctx.pubVolumeCreatePending.Unpublish(vcp.Key())
+		}
 	} else if d.destroy {
 		volumeCreated, fileLocation, err = destroyVolume(ctx, d.status)
+		if el, _ := ctx.pubVolumeCreatePending.Get(vcp.Key()); err == nil && el != nil {
```

Comment:
Why do you check for err == nil?
If the Get returns an object don't you always want to unpublish it?

Reply:
Good point. We are not worry about volume consistency here, so should delete. Will modify.